### PR TITLE
fix/merge config on dynamic config

### PR DESCRIPTION
**Pull request by CodeNoobKing**
#8 

## Summary by CodeRabbit

- **New Features**
	- Enhanced file copying capabilities during the build process with the introduction of `AdapterCopyService`.
	- Implemented strategies for copying different types of files, including class files and service directories, with specific handling for Spring factory configurations.
- **Refactor**
	- Optimized the method for adding artifacts to the project root by utilizing the new `AdapterCopyService`.
	- Removed outdated method for adding patches to the project root.
- **Tests**
	- Added comprehensive test coverage for new copying strategies and utilities, ensuring robust and error-free file operations.
- **Documentation**
	- Introduced example configuration files and test resources to demonstrate the new file copying and merging functionalities.

### CODE REVIEW

File: koupleless-build-plugin/koupleless-base-build-plugin/src/main/java/com/alipay/sofa/koupleless/base/build/plugin/adapter/ClassCopyStrategy.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    public void copy(File buildDir, String entryName, byte[] content) throws Throwable {
+        File fileToCreate = Paths.get(buildDir.getAbsolutePath(), entryName).toFile();
+        Files.createDirectories(fileToCreate.toPath().getParent());
+        if (!fileToCreate.exists()) {
+            Files.createFile(fileToCreate.toPath());
+        }
+
+        // byte to input stream
+        Files.write(fileToCreate.toPath(), content, StandardOpenOption.TRUNCATE_EXISTING);
+    }
```
[the remark this answers is not in the document]
Comment (CodeNoobKing) [reply koupleless-build-plugin/koupleless-base-build-plugin/src/main/java/com/alipay/sofa/koupleless/base/build/plugin/adapter/ClassCopyStrategy.java:42]: NONEED

File: koupleless-build-plugin/koupleless-base-build-plugin/src/main/java/com/alipay/sofa/koupleless/build/common/SpringUtils.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+public class SpringUtils {
+
+    private static SpringUtils instance = new SpringUtils();
+
+    public static SpringUtils INSTANCE() {
+        return instance;
+    }
+
+    @SneakyThrows
+    public Map<String, List<String>> parseSpringFactoryConfig(InputStream inputStream) {
```
Comment (lvjing2) [inline koupleless-build-plugin/koupleless-base-build-plugin/src/main/java/com/alipay/sofa/koupleless/build/common/SpringUtils.java:39]: 尽可能复用里面的方法 
 org.springframework.core.io.support.SpringFactoriesLoader#loadSpringFactories

Comment (CodeNoobKing) [reply koupleless-build-plugin/koupleless-base-build-plugin/src/main/java/com/alipay/sofa/koupleless/build/common/SpringUtils.java:41]: 参照了其逻辑

**Comment by lvjing2** [comment]
其他的 adapter 需要升级成该版本

**Pull request merged**